src/base/rss/rssdownloadrule.cpp

```diff
@@ -402,6 +468,36 @@ void DownloadRule::setEpisodeFilter(const QString &e)
     m_episodeFilter = e;
 }
 
+QStringList DownloadRule::previouslyMatchedEpisodes() const
+{
+    return m_previouslyMatchedEpisodes;
+}
+
+void DownloadRule::setPreviouslyMatchedEpisodes(const QStringList &previouslyMatchedEpisodes)
+{
+    m_previouslyMatchedEpisodes = previouslyMatchedEpisodes;
+}
+
+void DownloadRule::clearPreviouslyMatchedEpisodes()
+{
+    m_previouslyMatchedEpisodes.clear();
+}
+
+void DownloadRule::matched(const QString &article)
+{
+    if (m_smartFilter) {
+        QRegExp f(episodeRegex);
+
+        f.indexIn(article);
+        QString episodeStr = computeEpisodeName(f.cap(1));
+
+        if (!episodeStr.isEmpty()) {
+            //TODO: probably need to add a marker for PROPER/REPACK to avoid duplicate downloads
```

Comment:
The code here would not differ significantly from checks in `DownloadRule::matches()` then. Probably, it has to be refactored into a function.